lib/Core/Executor.cpp

```diff
@@ -4268,8 +4294,65 @@ void Executor::executeMemoryOperation(ExecutionState &state,
     if (incomplete) {
       terminateStateEarly(*unbound, "Query timed out (resolve).");
     } else {
-      terminateStateOnError(*unbound, "memory error: out of bound pointer", Ptr,
-                            NULL, getKValueInfo(*unbound, optimAddress));
+      if(LazyInitialization && !isWrite) {
+        const Array* array = nullptr;
+        if (!isPointer){
+          array = CreateArrayWithName(state, type, "lazy_init_arr");
+        }
+        ref<Expr> size = klee::ConstantExpr::alloc(type, Expr::Int64);
+        bool isLocal = false; // only allow the object to exist in the function
+        auto* mo = executeAlloc(state, size, isLocal, target, true);
+        auto* os = bindObjectInState(state, mo, isLocal, array);
+
+        uint64_t addressValue = 0;
+        if (isa<klee::ConstantExpr>(address.getValue())) {
+          addressValue = cast<klee::ConstantExpr>(address.getValue())->getZExtValue();
+        } else {
+          ref<ConstantExpr> result;
+          if (solver->getValue(state, address.getValue(), result)) {
+            addressValue = result->getZExtValue();
+          } else {
+            terminateStateOnError(*unbound, "memory error - lazy init: couldn't resolve address",
+                                  Ptr, nullptr, getKValueInfo(*unbound, optimAddress));
+            return;
+          }
+        }
+
+        auto segment = mo->getSegment();
+        if (isPointer) {
+          klee_message("LAZYINIT: this is a pointer! size: %u", bytes*8);
+        }
+        klee_message("LAZYINIT: segment: %lu address: %lu size:%lu", segment, addressValue, cast<klee::ConstantExpr>(size)->getZExtValue());
+
+        state.addressSpace.concreteAddressMap.insert({reinterpret_cast<uint64_t>(addressValue), segment});
```

Comment:
Why the new pointers are concrete? I guess you reworked this later -- then there are two options:
  - either you want to keep this in the history because of documentation purposes (like: hey, we tried this and that didn't work, because XXX), but then you must properly document in commit messages why that didn't work, or
  - just drop this code from the history. Break this commit into parts and keep only those that you use in the final code.